runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -4977,6 +4977,46 @@ J9::Z::TreeEvaluator::multianewArrayEvaluator(TR::Node * node, TR::CodeGenerator
    #undef iComment
    }
 
+
+///////////////////////////////////////////////////////////////////////////////////////
+// Generate code for multianewarray
+// Checks the number of dimensions. For 1 dimensional arrays call the helper, for >1 call
+// generateMultianewArrayWithInlineAllocators.
+///////////////////////////////////////////////////////////////////////////////////////
+TR::Register *
+J9::Z::TreeEvaluator::multianewArrayEvaluator(TR::Node * node, TR::CodeGenerator * cg)
+   {
+   TR::Compilation *comp = cg->comp();
+   TR_ASSERT_FATAL(comp->target().is64Bit(), "multianewArrayEvaluator is only supported on 64-bit JVMs!");
+
+   TR::Node *secondChild = node->getSecondChild();    // Number of dimensions - this is fixed in the bytecode, so compile time constant
+
+   // The number of dimensions should always be an iconst
+   TR_ASSERT_FATAL(secondChild->getOpCodeValue() == TR::iconst, "dims of multianewarray must be iconst");
+
+   // Only generate inline code if nDims > 1
+   uint32_t nDims = secondChild->get32bitIntegralValue();
+   if (nDims > 1)
+      {
+      return generateMultianewArrayWithInlineAllocators(node, cg);
+      }
+   else
+      {
+      // trace a message to indicate that inline allocation is disabled for nDims < 2
```

Comment:
Do we need a comment here? I think text in the Assert message is clear about disabling the inline allocation for multianewarray and also the comment on function makes it clear that for <1 dimensional array it calls helper.

Reply:
thanks, deleted